src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```
@@ -65,87 +66,81 @@
+     *   is URI to its associated FASTQ file on HDFS.
+     * @param recordLine    a line of text containing the breakpoint ID and URI to associated FASTQ file
+     * @return              a pair constructed by splitting the string into the ID part and the URI part
+     *                      the URI part will be null if cannot be parsed
      */
     @VisibleForTesting
-    static final class SGAAssemblyResult implements Serializable{
-        private static final long serialVersionUID = 1L;
-
-        public final ContigsCollection assembledContigs;
-        public final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> runtimeInformation;
-
-        public SGAAssemblyResult(final File contigFile, final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> collectedRuntimeInfo){
-            ContigsCollection proxy = null;
-            try{
-                proxy = new ContigsCollection(contigFile);
-            } catch (final IOException e) { // on IO error caused by converting contigs FASTA file to memory, set in-memory struct to null and append error message
-                collectedRuntimeInfo.add(new Tuple2<>(ContigsCollection.class.getName(),
-                                                      new Tuple3<>(ExternalCommandlineProgramModule.ReturnStatus.STDIOFAIL, "", e.getMessage())));
-            }
-            assembledContigs = proxy;
-            runtimeInformation = collectedRuntimeInfo;
+    static Tuple2<Long, URI> assignFASTQToBreakpoints(final String recordLine){
+        final String[] recordForOneBreakPoint = recordLine.split("\t");
+        final Long breakpointID = Long.valueOf(recordForOneBreakPoint[0]);
+        try{
+            final URI fastqURI = new URI(recordForOneBreakPoint[1]);
+            return new Tuple2<>(breakpointID, fastqURI);
+        }catch (final URISyntaxException e){
```

I don't think we should catch and swallow this exception here. We should probably re-throw it as a UserException (since it indicates that the input data was bad, containing improperly formed URI strings).